File: 待办清单功能/ruankai2/app/src/main/java/com/byted/camp/todolist/operation/db/FeedReaderContract.java
package com.byted.camp.todolist.operation.db;

import android.provider.BaseColumns;


public class FeedReaderContract {

    public static final String SQL_CREATE_ENTRIES =
            "CREATE TABLE " + FeedEntry.TABLE_NAME + " (" +
                    FeedEntry._ID + " INTEGER PRIMARY KEY," +
                    FeedEntry.COLUMN_NAME_TITLE + " TEXT," +
                    FeedEntry.COLUMN_NAME_SUBTITLE + " TEXT)";

    public static final String SQL_DELETE_ENTRIES = "DROP TABLE IF EXISTS " + FeedEntry.TABLE_NAME;

    // 防止有人不慎将抽象类实例化
    // 将构造方法设置成private
    private FeedReaderContract() {
    }

    /* Inner class that defines the table contents */
    public static class FeedEntry implements BaseColumns {

        public static final String TABLE_NAME = "entry";

        public static final String COLUMN_NAME_TITLE = "title";

        public static final String COLUMN_NAME_SUBTITLE = "subtitle";

        public static final String COLUMN_NAME_CONTENT = "content";
    }
}

File: 12.25整合代码/MyApplication2/app/src/main/java/com/example/myapplication/todolist/operation/db/FeedReaderContract.kt
package com.example.myapplication.todolist.operation.db

import android.provider.BaseColumns

object FeedReaderContract {
    const val SQL_CREATE_ENTRIES = "CREATE TABLE " + FeedEntry.TABLE_NAME + " (" +
            BaseColumns._ID + " INTEGER PRIMARY KEY," +
            FeedEntry.COLUMN_NAME_TITLE + " TEXT," +
            FeedEntry.COLUMN_NAME_SUBTITLE + " TEXT)"
    const val SQL_DELETE_ENTRIES = "DROP TABLE IF EXISTS " + FeedEntry.TABLE_NAME

    /* Inner class that defines the table contents */
    object FeedEntry : BaseColumns {

        const val TABLE_NAME = "entry"
        const val COLUMN_NAME_TITLE = "title"
        const val COLUMN_NAME_SUBTITLE = "subtitle"
        const val COLUMN_NAME_CONTENT = "content"


    }
}